samples/nrf9160/modem_shell/src/main.c

```diff
@@ -142,25 +151,26 @@ void nrf_modem_fault_handler(struct nrf_modem_fault_info *fault_info)
 static void mosh_print_version_info(void)
 {
 #if defined(APP_VERSION)
-	printk("\nMOSH version:       %s", STRINGIFY(APP_VERSION));
+	mosh_print("\nMOSH version:       %s", STRINGIFY(APP_VERSION));
 #else
-	printk("\nMOSH version:       unknown");
+	mosh_print("\nMOSH version:       unknown");
 #endif
 
 #if defined(BUILD_ID)
-	printk("\nMOSH build id:      v%s", STRINGIFY(BUILD_ID));
+	mosh_print("MOSH build id:      v%s", STRINGIFY(BUILD_ID));
 #else
-	printk("\nMOSH build id:      custom");
+	mosh_print("MOSH build id:      custom");
 #endif
 
 #if defined(BUILD_VARIANT)
 #if defined(BRANCH_NAME)
-	printk("\nMOSH build variant: %s/%s\n\n", STRINGIFY(BRANCH_NAME), STRINGIFY(BUILD_VARIANT));
+	mosh_print("MOSH build variant: %s/%s\n\n",
```

Comment:
I discovered that with the LwM2M carrier lib overlay, the shell doesn't get started before mosh tries to print the version information. The shell backend is initialized using `SYS_INIT()`, but it creates a new thread and shell really gets started only when that thread gets to run for the first time. This is probably the reason why we have used `printk()` in `main()` earlier... Because of this, I restored `main()` to use `printk()`.